-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e3568/32439/summary.html
**COMMIT**: 50be075c9642ac567f27259c469618195e5cacf3
**CMSSW**: CMSSW_13_1_X_2023-05-06-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/41573/32439/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_1_X_2023-05-06-1100/src/DataFormats/L1THGCal/src/HGCalTower.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_1_X_2023-05-06-1100/src/DataFormats/L1THGCal/src/HGCalTowerMap.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_1_X_2023-05-06-1100/src/DataFormats/L1THGCal/src/HGCalTriggerCell.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_1_X_2023-05-06-1100/src/DataFormats/L1THGCal/src/HGCalTriggerSums.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_1_X_2023-05-06-1100/src/DataFormats/L1THGCal/src/HGCalTowerMap.cc: In member function 'const l1t::HGCalTowerMap& l1t::HGCalTowerMap::operator+=(const l1t::HGCalTowerMap&)':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_1_X_2023-05-06-1100/src/DataFormats/L1THGCal/src/HGCalTowerMap.cc:18:12: error: 'endl' was not declared in this scope; did you mean 'std::endl'?
   18 |         << endl;
      |            ^~~~
      |            std::endl
In file included from /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02783/el8_amd64_gcc11/external/gcc/11.2.1-f9b9dfdd886f71cd63f5538223d8f161/include/c++/11.2.1/istream:39,
                 from /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02783/el8_amd64_gcc11/external/gcc/11.2.1-f9b9dfdd886f71cd63f5538223d8f161/include/c++/11.2.1/sstream:38,
</pre>

--- a/DataFormats/L1Trigger/interface/BXVector.h
+++ b/DataFormats/L1Trigger/interface/BXVector.h
@@ -1,5 +1,5 @@
-#ifndef BXVector_h
-#define BXVector_h
+#ifndef DataFormats_L1Trigger_BXVector_h
+#define DataFormats_L1Trigger_BXVector_h
 
 // this class is an extension of std::vector
 // designed to store objects corresponding to several time-samples (BX)
@@ -134,4 +134,4 @@ private:
 
 #include "BXVector.icc"
 
-#endif
+#endif  // DataFormats_L1Trigger_BXVector_h

--- a/DataFormats/L1Trigger/interface/BXVector.icc
+++ b/DataFormats/L1Trigger/interface/BXVector.icc
@@ -1,6 +1,8 @@
+#include <cassert>
+
+// FIXME: these 3 lines are required by other packages
 #include <vector>
 #include <iostream>
-#include <cassert>
 using namespace std;
 
 template <class T>
@@ -211,15 +213,27 @@ unsigned BXVector<T>::indexFromBX(int bx) const {
 // check to see if bx is empty
 template <class T>
 bool BXVector<T>::isEmpty(int bx) const {
-  if (itrs_[indexFromBX(bx)] == data_.size()) {
+  if (bx < bxFirst_) {
     return true;
   }
 
-  if (indexFromBX(bx + 1) >= itrs_.size()) {
+  auto const index_bx = indexFromBX(bx);
+
+  if (index_bx >= itrs_.size()) {
+    return true;
+  }
+
+  if (itrs_[index_bx] == data_.size()) {
+    return true;
+  }
+
+  auto const index_bxPlus1 = indexFromBX(bx + 1);
+
+  if (index_bxPlus1 >= itrs_.size()) {
     return false;
   }
 
-  if (itrs_[indexFromBX(bx)] == itrs_[indexFromBX(bx + 1)]) {
+  if (itrs_[index_bx] == itrs_[index_bxPlus1]) {
     return true;
   }
 

